spec.md

```diff
@@ -1957,6 +2319,72 @@ asClock(x)
 [@sec:primitive-operations] will describe the format and semantics of each
 primitive operation.
 
+## Reading Probe References
+
+Probes are read using the `read`{.firrtl} operation.
+
+Read expressions have source flow and can be connected to other components:
+
+```firrtl
+module Foo :
+  output p : Probe<UInt>
+  ; ...
+
+module Bar :
+  output x : UInt
+
+  inst f of Foo
+  x <= read(f.p) ; indirectly access the probed data
+```
+
+Indexing statically (sub-field, sub-index) into a probed value is allowed as
+part of the read:
+
+```firrtl
+module Foo :
+  output p : Probe<{a: UInt, b: UInt}>
+  ; ...
+
+module Bar :
+  output x : UInt
+
+  inst f of Foo
+  x <= read(f.p.b) ; indirectly access the probed data
+```
+
+The source of the probe must reside at or below the point of the `read`
+expression in the design hierarchy.
+
+## Probe
```

Comment:
These probe expressions seem unnecessary to me - is there a reason why they are not just part of an `export` statement? If there is a reason, an example demonstrating the use-case behind this difference would be helpful.

Reply:
I'm considering collapsing export and forward into a single statement now that they're both `ref<T> -> ref<T>` and do the same thing otherwise re:statically indicating the direction references are routed.   This is mostly stuck on picking solid syntax/name.  In the tentative/WIP CIRCT implementation both of these are currently parsed into a `firrtl.ref.assign` op, which is kinda connect-like (afraid to say that, but hopefully YKWIM) but no last-connect semantics (error to target same dest more than once, not turned into conditional connections if under a when, so on).

If not, probably can (should?) collapse `export probe(r) as t` and `export rwprobe(r) as t` into, e.g., `export ro r as t` and `export rw r as t` (or however we like, point is don't have separate expression for generating the reference) since as you've noticed `probe` expressions only have the one use presently.  In the future we may find the separation useful (otherwise no way to generate a reference without directly routing it to a port), but it's maybe a little unnecessary as -is since that's all you can do with it anyway.

It's important to specify the difference between whether you want a `Probe` or a `RWProbe`, otherwise it'd just be `export r as t`, if that wasn't clear.

WDYT / make sense?

Reply:
But doesn't `r` define the type, as either a `RWProbe` or a `Probe`?

Reply:
The (perhaps limited) utility of the `probe` / `rwprobe` expressions is exactly that-- it's what decides/indicates if we have a `Probe` or `RWProbe` type.  Presently they're the only way to originate a `Probe` or `RWProbe` at all.

That's why `r` must be a `probe` or `rwprobe` expression in the `export` statement as currently proposed.